eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -313,4 +345,80 @@ public function extractFilesFromXml($xml)
 
         return null;
     }
+
+    public function getAllImagesData(): array
+    {
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select(
+                $this->connection->quoteIdentifier('contentobject_attribute_id'),
+                $this->connection->quoteIdentifier('filepath')
+            )->from($this->connection->quoteIdentifier(self::IMAGE_FILE_TABLE))
+            ->distinct()
+        ;
+
+        return $selectQuery->execute()->fetchAllAssociative();
+    }
+
+    public function updateImageData(int $fieldId, int $versionNo, string $xml): void
+    {
+        $updateQuery = $this->connection->createQueryBuilder();
+        $expressionBuilder = $updateQuery->expr();
+        $updateQuery
+            ->update(
+                $this->connection->quoteIdentifier('ezcontentobject_attribute')
+            )
+            ->set(
+                $this->connection->quoteIdentifier('data_text'),
+                ':xml'
+            )
+            ->where(
+                $expressionBuilder->eq(
+                    $this->connection->quoteIdentifier('id'),
+                    ':fieldId'
+                )
+            )
+            ->andWhere(
+                $expressionBuilder->eq(
+                    $this->connection->quoteIdentifier('version'),
+                    ':versionNo'
+                )
+            )
+            ->setParameter(':fieldId', $fieldId, PDO::PARAM_INT)
+            ->setParameter(':versionNo', $versionNo, PDO::PARAM_INT)
+            ->setParameter(':xml', $xml, PDO::PARAM_STR)
+            ->execute()
+        ;
+    }
+
+    public function updateImagePath(int $fieldId, string $oldPath, string $newPath): void
```

Comment:
The method should be declared on `\eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway`